src/psyclone/transformations.py

```diff
@@ -2630,6 +2676,207 @@ def validate(self, kern):
                                       format(kern.name))
 
 
+class ACCKernelsTrans(RegionTrans):
+    '''
+    Enclose a sub-set of nodes from a Schedule within an OpenACC kernels
+    region (i.e. within "!$acc kernels" ... "!$acc end kernels" directives).
+    Currently only supported for the NEMO API.
+
+    For example:
+
+    >>> from psyclone.parse import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api = "NEMO"
+    >>> filename = "tra_adv.F90"
+    >>> ast, invokeInfo = parse(filename, api=api)
+    >>> psy = PSyFactory(api).create(invokeInfo)
+    >>>
+    >>> from psyclone.transformations import ACCKernelsTrans
+    >>> ktrans = ACCKernelsTrans()
+    >>>
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>> kernels = schedule.children[0].children[0].children[0:-1]
+    >>> # Transform the kernel
+    >>> new_sched, _ = ktrans.apply(kernels)
+
+    '''
+    from psyclone import nemo, psyGen
+    valid_node_types = (nemo.NemoLoop, nemo.NemoKern, nemo.NemoIfBlock,
```

Comment:
This is repeated later.

Reply:
Apologies if I'm being dim but what's repeated?